### PR TITLE
Leonardo fontes

### PR DESCRIPTION
**Pull request by leofontes**
Boa tarde pessoal,

Mandando meu desafio, fico no aguardo por feedbacks. Valeu!

### CODE REVIEW

File: project/js/interactions.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			arr[j + 1] = arr[j];
+		}
+
+		arr[j + 1] = tmpObj;
+	}
+
+	//(EN) Return the top5 
+	//(PT) Retorna o top5
+	return [arr[arr.length - 1], arr[arr.length - 2], arr[arr.length - 3], arr[arr.length - 4], arr[arr.length - 5]];
+}
```
Comment (brunoventura) [inline project/js/interactions.js]: There are some issues here: 
1. The method name is sortArr, that means this function is to sort an array, but if I call this method I got my array sorted and the top 5.
2. If the given array has length < 5, your code will break.

Try to use the language benefits, javascript have a sort method that uses a mix of insertion sort and quick sort:
https://github.com/v8/v8/blob/master/src/js/array.js#L712

And slice, to get a part of an array:
http://www.w3schools.com/jsref/jsref_slice_array.asp

File: project/js/interactions.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		});
+
+	});
+
+});
+
+//(EN) Insertion sort adapted to our array of objects, using the hits as criteria
+//(PT) Insertion sort adaptado para a array de objetos, usando como critério de ordenação os hits
+function sortArr(arr) {
+	for(let i = 0; i < arr.length; i++) {
```
Comment (brunoventura) [inline project/js/interactions.js]: I'ts a good practice put all functions inside the the ready (if you are using jQuery) to not pollute the global scope

File: project/js/interactions.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		$('#rankingNumberThree').text(topFive[2].hits);
+		//(EN) Change values for #4
+		//(PT) Muda conteúdo do #4
+		$('#rankingLinkFour').text(topFive[3].shortUrl);
+		$('#rankingNumberFour').text(topFive[3].hits);
+		//(EN) Change values for #5
+		//(PT) Muda conteúdo do #5
+		$('#rankingLinkFive').text(topFive[4].shortUrl);
+		$('#rankingNumberFive').text(topFive[4].hits);
+	});
```
Comment (brunoventura) [inline project/js/interactions.js]: Try to make this inside an iteration, and not use fixed index from array, nor use fixed ids form html. Dynamicity is one of the major features of programming.

File: project/index.html [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	</section>
+	<!--(EN) Number of hits -->
+	<!--(PT) Número de hits -->
+	<section class="hits-section">
+		<div class="arrow"></div>
+		<div class="container">
+			<div class="hits-container">
+				<h2 class="hits-title">HITS</h2>
+
+				<p><span class="hits-counter">35.713.571</span></p>
```
Comment (brunoventura) [inline project/index.html]: Are you recalculating this hit counter in some place?

File: project/index.html [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+					<div class="col-sm-4"><p class="ranking-number" id="rankingNumberThree">2111</p></div>
+				</div>
+				<div class="row">
+					<div class="col-sm-8"><p class="ranking-link" id="rankingLinkFour">http://chr.dc/fghfgh</p></div>
+					<div class="col-sm-4"><p class="ranking-number" id="rankingNumberFour">617</p></div>
+				</div>
+				<div class="row">
+					<div class="col-sm-8"><p class="ranking-link" id="rankingLinkFive">http://chr.dc/yuiyui</p></div>
+					<div class="col-sm-4"><p class="ranking-number" id="rankingNumberFive">43</p></div>
+				</div>
```
Comment (brunoventura) [inline project/index.html]: Can you do the logic without this fake data?

File: project/index.html (hunk 4 above)
Comment (leofontes) [inline project/index.html]: Sorry, that was my bad, I focused on the Top 5 section and completely forgot to add them up and substitute there.

File: project/js/interactions.js (hunk 1 above)
Comment (leofontes) [inline project/js/interactions.js]: Totally agree 👍

File: project/js/interactions.js (hunk 2 above)
Comment (leofontes) [inline project/js/interactions.js]: Didn't know that, thank you for pointing it out 😃

File: project/index.html (hunk 5 above)
Comment (leofontes) [inline project/index.html]: Only appending the structure if there is data on the JSON? Yes I think so..

**Comment by brunoventura** [comment]
Hi @leofontes, thanks for submitting your PR.
My revision is more focused in the javascript code and good practices, other person will evaluate your html semantic and css skills
I have pointed some pieces of code like we do here at chaordic, so if you wish you can rewrite these and make new commits, so I can reevaluate.

File: project/index.html (hunk 5 above)
Comment (brunoventura) [inline project/index.html]: Yes :) make it more dynamic

File: project/index.html [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+(PT)
+Autor: Leonardo Fontes
+Data: Quarta-feira, 13 de Julho de 2016
+-->
+<html>
+<head>
+	<title>Chaordic Challenge</title>
+	<meta charset="UTF-8">
+	<link href='https://fonts.googleapis.com/css?family=Roboto+Slab:400,700|Roboto:400,700' rel='stylesheet' type='text/css'>
+	<link rel="stylesheet" type="text/css" href="styles/css/bootstrap.min.css">
```
Comment (diogomoretti) [inline project/index.html]: Vejo que não havia a necessidade de importar o bootstrap inteiro consumindo 118 kb sendo que você só usou o grid bem dizer... que aliás, também não precisa de grid para uma página dessas =]

File: project/index.html [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+					<div class="col-sm-8"><p class="ranking-link" id="rankingLinkFive">http://chr.dc/yuiyui</p></div>
+					<div class="col-sm-4"><p class="ranking-number" id="rankingNumberFive">43</p></div>
+				</div>
+			</div>
+		</div>
+	</section>
+	<!--(EN) Number of hits -->
+	<!--(PT) Número de hits -->
+	<section class="hits-section">
+		<div class="arrow"></div>
```
Comment (diogomoretti) [inline project/index.html]: Você poderia fazer essa seta usando [pseudo-elements](https://css-tricks.com/almanac/selectors/a/after-and-before/) =]

File: project/styles/sass/home.scss [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+.shorten-button {
+	@include roboto-bold;
+	background-color: $orange-inputs;
+	color: $white;
+	cursor: pointer;
+	margin-top: 12%;
+	padding: 5%;
+	pointer: cursor;
```
Comment (diogomoretti) [inline project/styles/sass/home.scss]: Aqui está invertido e também já foi declarado 3 linhas acima =]

**Comment by diogomoretti** [comment]
@leofontes obrigado pelo seu PR =] gostei bastante da semântica do seu html e também da escrita das classes. só aqueles pontos ali de comentei 👍

File: project/index.html (hunk 6 above)
Comment (leofontes) [inline project/index.html]: Concordo! Acabei usando o Bootstrap como muleta mas dava de caminhar sozinho haha 😄

File: project/index.html (hunk 7 above)
Comment (leofontes) [inline project/index.html]: Boa ideia, não me ocorreu, realmente ficaria melhor.

File: project/styles/sass/home.scss (hunk 8 above)
Comment (leofontes) [inline project/styles/sass/home.scss]: Viajei 💩 Obrigado pelo feedback Diogo, vou corrigir e mandar junto com as mudanças que o Bruno sugeriu.

File: project/index.html (hunk 4 above)
Comment (leofontes) [inline project/index.html]: Should I also append the hits section only if there are hits, or should it always be there and if there are no hits just show 0?

Comment (brunoventura) [inline project/index.html]: We not specify this, but IMHO it's better to keep 0

**Comment by leofontes** [comment]
@brunoventura e @diogomoretti implementei as sugestões de vocês, obrigado pelo feedback.

**Comment by diogomoretti** [comment]
Olá @leofontes. Mais uma vez obrigado pelo seu PR. Vamos fechar aqui e dar continuidade pelo RecruiterBox =]

**Pull request closed without merging**